org.lflang/src/org/lflang/generator/PortInstance.java

```diff
@@ -0,0 +1,616 @@
+STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***************/
+package org.lflang.generator;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.PriorityQueue;
+import java.util.Set;
+
+import org.lflang.ErrorReporter;
+import org.lflang.lf.Input;
+import org.lflang.lf.Output;
+import org.lflang.lf.Parameter;
+import org.lflang.lf.Port;
+import org.lflang.lf.WidthSpec;
+import org.lflang.lf.WidthTerm;
+
+/** 
+ * Representation of a runtime instance of a port.
+ * This may be a single port or a multiport.
+ *  
+ * @author{Marten Lohstroh <marten@berkeley.edu>}
+ * @author{Edward A. Lee <eal@berkeley.edu>}
+ */
+public class PortInstance extends TriggerInstance<Port> {
+
+    /**
+     * Create a runtime instance from the specified definition
+     * and with the specified parent that instantiated it.
+     * @param instance The Instance statement in the AST.
+     * @param parent The parent.
+     */
+    public PortInstance(Port definition, ReactorInstance parent) {
+        this(definition, parent, null);
+    }
+
+    /**
+     * Create a runtime instance from the specified definition
+     * and with the specified parent that instantiated it.
+     * @param instance The Instance statement in the AST.
+     * @param parent The parent.
+     * @param errorReporter An error reporter, or null to throw exceptions.
+     */
+    public PortInstance(Port definition, ReactorInstance parent, ErrorReporter errorReporter) {
+        super(definition, parent);
+        
+        if (parent == null) {
+            throw new NullPointerException("Cannot create a PortInstance with no parent.");
+        }
+        
+        // If this is a multiport, determine the width.
+        WidthSpec widthSpec = definition.getWidthSpec();
+
+        if (widthSpec != null) {
+            if (widthSpec.isOfVariableLength()) {
+                errorReporter.reportError(definition,
+                        "Variable-width multiports not supported (yet): " + definition.getName());
+            } else {
+                isMultiport = true;
+                
+                // Determine the initial width, if possible.
+                // The width may be given by a parameter or even sum of parameters.
+                width = 0;
+                for (WidthTerm term : widthSpec.getTerms()) {
+                    Parameter parameter = term.getParameter();
+                    if (parameter != null) {
+                        Integer parameterValue = parent.initialIntParameterValue(parameter);
+                        // Only a Literal is supported.
+                        if (parameterValue != null) {
+                            // This could throw NumberFormatException
+                            width += parameterValue;
```

Comment:
I think for the following comment to be accurate, this statement should perhaps be in a `try...catch`, with catch using the provided error reporter, if any:
> An error reporter, or null to throw exceptions.

Reply:
Thanks. Comment is obsolete. Removed.